### ISSUE TITLE
Post Settings: Redesign the Category Picker

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#5810]
The current version of the Category Picker is quite ugly. 

Step 1: use the material grid to align everything (the tree view and the "add new category" dialog). And fix the incoherent font sizes.
Step 2: make that tree view more androidy - update the indicator graphic, we might get some inspiration from https://material.io/guidelines/components/lists.html

**Comment by maxme** [comment]
This is current design in wpios:

<img width="618" alt="screen shot 2017-05-04 at 17 36 19" src="https://cloud.githubusercontent.com/assets/40213/25713046/a65507d6-30f3-11e7-96c1-1cac22cfb82c.png">


And Calypso:

<img width="417" alt="screen shot 2017-05-04 at 18 00 28" src="https://cloud.githubusercontent.com/assets/40213/25713052/ac0c682c-30f3-11e7-844f-90466ff2f69e.png">

**Comment by maxme** [comment]
Category looks better after a first cleanup:
![category_picker](https://cloud.githubusercontent.com/assets/40213/26732672/b1d678f2-47b8-11e7-9822-7bb52647e834.png)


I used the margins/sizes from the [Material Specs for lists](https://material.io/guidelines/components/lists.html#lists-specs), which are:

```
Font: Roboto Regular 16sp
Height: 48dp
Left and right padding from screen edge: 16dp
Top padding: 8dp
```

I also used the same divider and background color as other list-like screens (divider: 1dp grey_lighten_30 on white background).

@mattmiklic, other ideas from improving this are welcome :).

**Comment by maxme** [comment]
I just updated the "create category dialog", using [Material Specs for dialogs](https://material.io/guidelines/components/dialogs.html#dialogs-specs).

![screenshot-2017-06-09_16 18 18 360](https://user-images.githubusercontent.com/40213/26979896-bef22c02-4d30-11e7-97d1-70f4dbd95bbe.png)

**Comment by mattmiklic** [comment]
> Category picker looks better after a first cleanup:
> @mattmiklic, other ideas from improving this are welcome :).

This looks good to me, a definite improvement over what's there. I'd ship this.

> I just updated the "create category dialog", using Material Specs for dialogs.

Two comments on this one -- first, the orange button text isn't consistent with what we're doing elsewhere, like in Site Settings, where the text buttons are blue_medium. Second, I think the category item (I don't know the name for this kind of control in Android) should have an underline below it -- in the Material Design spec, items that open dropdown menus are styled similarly to text fields, with the addition of the downward-facing arrow on the right-hand side.

<img width="760" alt="components_menus_behavior2" src="https://user-images.githubusercontent.com/1396552/27191737-c106bda6-51c6-11e7-8a0d-182fce28ad31.png">

**Comment by oguzkocer** [comment]
Just adding a note here that the blue button should be fixed in #6058. I've changed all the AlertDialogs in Post Settings to use the Calypso style which uses the blue button.

**Comment by maxme** [comment]
![screenshot-2017-06-16_10 43 55 644](https://user-images.githubusercontent.com/40213/27219216-3576a204-5281-11e7-9b3f-13996a2f8e2d.png)

@oguzkocer @mattmiklic I confirm the button colors are correct now. I'm keeping this issue opened to fix the missing underline under the spinner.
